src/main/kotlin/compiler/ast/N_LITERAL.kt

```diff
@@ -19,32 +19,46 @@ abstract class N_LITERAL: N_EXPR() {
 class N_LITERAL_BOOLEAN(val value: Boolean): N_LITERAL() {
     override fun toText() = if (value) "true" else "false"
     override fun codeValue(coder: Coder) { coder.value(this, value) }
+    override fun constantValue() = VBool(value)
 }
 
 class N_LITERAL_INTEGER(val value: Int): N_LITERAL() {
     override fun toText() = "$value"
     override fun codeValue(coder: Coder) { coder.value(this, value) }
+    override fun constantValue() = VInt(value)
 }
 
 class N_LITERAL_FLOAT(val value: Float): N_LITERAL() {
     override fun toText() = "$value"
     override fun codeValue(coder: Coder) { coder.value(this, value) }
+    override fun constantValue() = VFloat(value)
 }
 
 class N_LITERAL_OBJ(val objID: ObjID): N_LITERAL() {
     override fun toText() = "#$id"
     override fun codeValue(coder: Coder) { coder.value(this, objID) }
+    override fun constantValue() = VObj(objID)
 }
 
 class N_LITERAL_STRING(val value: String): N_LITERAL() {
     override fun toText() = "\"$value\""
     override fun codeValue(coder: Coder) { coder.value(this, value) }
+    override fun constantValue() = VString(value)
 }
 
 class N_LITERAL_LIST(val value: List<N_EXPR>): N_LITERAL() {
     override fun kids() = value
     override fun toText() = value.joinToString(", ", "LIST[", "]")
+    override fun constantValue(): Value? {
+        val constant = mutableListOf<Value>()
+        value.forEach { expr ->
+            val exprConstant = expr.constantValue()
+            if (exprConstant == null) return null else constant.add(exprConstant)
```

Comment:
Does returning null here do what you intended?  I think maybe you want to: 
```
for (expr in value) {
    val exprConstant = expr.constantValue()
    if (exprConstant == null) return null else constant.add(exprConstant)
}
```

Reply:
AAARGH.  Damn you Kotlin!!